.scripts/deploynaut-php-resque

```diff
@@ -79,6 +80,27 @@ do_stop()
 	return "$RETVAL"
 }
 
+#
+# Does some checks before stopping to ensure no tasks are running.
+# This will force a loop until no tasks are running.
+#
+pre_stop_check()
+{
+	# We need to ensure we have no pending or working tasks.
+	bash $STATUSFILE | egrep "^(pending|working):\ [1-9]"
+
+	# if egrep finds something it means a task is working or pending. In this it will
+	# return a 0 exit code. If we get a 0 exit code, we need to prevent the service from stopping.
+	if [ $? -eq 0 ]
+	then
+		echo "Jobs are running. Unable to stop service."
+		echo "Sleeping for 10 seconds."
+		sleep 10
+		pre_stop_check
+	fi
```

Comment:
If a deployment is started within the 10 second delay, this script would not be stopped until that deployment had also finished. If a deployment was started in the millisecond between pre_stop_check and do_stop, I doubt it would get very far :)